### PR TITLE
Hive partition overwrite by default

### PR DESCRIPTION
**Pull request by azafty468**
Fix for DE-490

### CODE REVIEW

File: edx/analytics/tasks/util/hive.py [hunk 1]
```diff
@@ -426,6 +426,13 @@ class OverwriteAwareHiveQueryDataTask(WarehouseMixin, OverwriteOutputMixin, Hive
     A generalized Data task whose output is a hive table populated from a hive query.
     """
 
+    overwrite_target_partition = Parameter(
```
Comment (brianhw) [inline edx/analytics/tasks/util/hive.py]: Is there a reason this isn't a BooleanParameter?

File: edx/analytics/tasks/util/hive.py [hunk 2]
```diff
@@ -437,18 +444,26 @@ def hive_partition_task(self):
...
+        if self.overwrite_target_partition:
+            return "OVERWRITE"
+        else:
+            return "INTO"
+
     def query(self):  # pragma: no cover
         full_insert_query = """
                     USE {database_name};
-                    INSERT INTO TABLE {table}
+                    INSERT {overwrite_check} TABLE {table}
```
Comment (brianhw) [inline edx/analytics/tasks/util/hive.py]: nit:  small preference for "into_or_overwrite" or similar for variable name here.   What's being inserted is not a "_check"...

File: edx/analytics/tasks/util/url.py [hunk 3]
```diff
@@ -37,6 +44,20 @@ def touch_marker(self):  # pragma: no cover
...
+
+            while read_attempts > 0 and not marker_exists:
+                marker_exists = self.exists()
+
+                if not marker_exists:
+                    log.debug("Marker file %s does not exist, sleeping for 60 seconds", marker)
+                    time.sleep(60)
+
+            if not marker_exists:
+                log.error("Error Marker file %s should have been created but could not be read!", marker)
```
Comment (brianhw) [inline edx/analytics/tasks/util/url.py:59]: Do we want this just to emit an error, or should it also raise an exception?

**Review by brianhw — COMMENTED** [review]
Looks good -- a few things to clarify.

File: edx/analytics/tasks/util/hive.py (hunk 1 above)
Comment (azafty468) [reply edx/analytics/tasks/util/hive.py]: Nope, no good reason at all.  Changing now

File: edx/analytics/tasks/util/hive.py (hunk 2 above)
Comment (azafty468) [reply edx/analytics/tasks/util/hive.py]: Yup, I'll adjust

File: edx/analytics/tasks/util/url.py (hunk 3 above)
Comment (azafty468) [reply edx/analytics/tasks/util/url.py:59]: I'm a little bit weary to raise an exception and impact a job if we haven't actually seen a problem before.

**Review by brianhw — APPROVED** [review]
LGTM.  👍

**Review by rao-abdul-mannan — APPROVED** [review]
Looks good.

**Pull request merged**